packages/field/src/FormatMixin.js

```diff
@@ -300,6 +358,18 @@ export const FormatMixin = dedupeMixin(
         this._reflectBackFormattedValueToUser();
       }
 
+      // Needed for choice-inputs and select (IE11) as an alternative to the input event.
+      // TODO: add in FormatMixin and remove here and in lion-select
+      _onChange() {
+        if (super._onChange) super._onChange();
+        this.dispatchEvent(
+          new CustomEvent('user-input-changed', {
+            bubbles: true,
+            composed: true,
```

Comment:
you didn't only copy, but also added `composed: true`

this is not needed, and adding it for some consistency is a bad idea, because you create an extra event which will bubble outside of shadow root while this is not expected by developers... so this fix to consistency creates additional problems without the real need for it

for now I would just leave it as it was before (without `composed: true`) and in the long run revisit all events we dispatch and double check if they need to bubble and bubble outside of shadow roots or not, and the decision in each case will be individual based on our API needs and users needs

Reply:
Fair point again.
I will do an attempt to document everything Event related:
- no CustomEvents
- bubble only when needed
- composed only when needed

Reply:
@tlouisse ha-ha, add one more :D https://twitter.com/JonathanAschan/status/1138744064835280897